src/Abstractions/NexusMods.Abstractions.Loadouts/Files/File.cs

```diff
@@ -0,0 +1,89 @@
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Loadouts.Ids;
+using NexusMods.Abstractions.Loadouts.Mods;
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.MnemonicDB.Attributes.Extensions;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+
+namespace NexusMods.Abstractions.Loadouts.Files;
+
+/// <summary>
+/// Represents an individual file which belongs to a <see cref="Loadout"/>, all files
+/// should at least have the <see cref="Loadout"/> reference, and optionally a reference to a <see cref="Mod"/>,
+/// </summary>
+public static class File
+{
+    private const string Namespace = "NexusMods.Abstractions.Loadouts.Mods.ModFile";
+
+    /// <summary>
+    /// The loadout this file is part of.
+    /// </summary>
+    public static readonly ReferenceAttribute Loadout = new(Namespace, nameof(Loadout));
+    
+    /// <summary>
+    /// The mod this file belongs to, if any.
+    /// </summary>
+    public static readonly ReferenceAttribute Mod = new(Namespace, nameof(Mod));
+    
+    /// <summary>
+    /// The location the file will be installed to
+    /// </summary>
+    public static readonly GamePathAttribute To = new(Namespace, nameof(To));
+    
+    /// <summary>
+    /// Standard model for a file.
+    /// </summary>
+    /// <param name="tx"></param>
+    public class Model(ITransaction tx) : Entity(tx)
+    {
+
+        /// <summary>
+        /// Gets the file id
+        /// </summary>
+        public FileId FileId => FileId.From(Id);
+
+        public EntityId LoadoutId
+        {
+            get => File.Loadout.Get(this);
+            set => File.Loadout.Add(this, value);
+        }
```

Comment:
`Loadout.LoadoutId` is of type `LoadoutId` while both `File.LoadoutId` and `Mod.LoadoutId` return an `EntityId`. This inconsistency is also present with `ModId` in both `File` and `Mod` models. This seems like something that'll easily result in more inconsistencies and weird behavior. We have `FileId`, `ModId`, and `LoadoutId` for a reason, so we should probably use them.

Reply:
Same with the Mod's LoadoutId